File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/Diagnostic/Test/UnitTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;
using VerifyCS = Microsoft.CodeAnalysis.CSharp.Testing.MSTest.CodeFixVerifier<
    $saferootprojectname$.$saferootidentifiername$Analyzer,
    $saferootprojectname$.CSharp$saferootidentifiername$CodeFixProvider>;
using VerifyVB = Microsoft.CodeAnalysis.VisualBasic.Testing.MSTest.CodeFixVerifier<
    $saferootprojectname$.$saferootidentifiername$Analyzer,
    $saferootprojectname$.VisualBasic$saferootidentifiername$CodeFixProvider>;

namespace $safeprojectname$
{
    [TestClass]
    public class $saferootidentifiername$UnitTest
    {
        //No diagnostics expected to show up
        [TestMethod]
        public async Task TestMethod1_CSharp()
        {
            var test = @"";

            await VerifyCS.VerifyAnalyzerAsync(test);
        }

        [TestMethod]
        public async Task TestMethod1_VisualBasic()
        {
            var test = @"";

            await VerifyVB.VerifyAnalyzerAsync(test);
        }

        //Diagnostic and CodeFix both triggered and checked for
        [TestMethod]
        public async Task TestMethod2_CSharp()
        {
            var test = @"
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    using System.Diagnostics;

    namespace ConsoleApplication1
    {
        class {|#0:TypeName|}
        {   
        }
    }";

            var fixtest = @"
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    using System.Diagnostics;

    namespace ConsoleApplication1
    {
        class TYPENAME
        {   
        }
    }";

            var expected = VerifyCS.Diagnostic("$saferootidentifiername$").WithLocation(0).WithArguments("TypeName");
            await VerifyCS.VerifyCodeFixAsync(test, expected, fixtest);
        }

        [TestMethod]
        public async Task TestMethod2_VisualBasic()
        {
            var test = @"
Class {|#0:TypeName|}

    Sub Main()

    End Sub

End Class";

            var fixtest = @"
Class TYPENAME

    Sub Main()

    End Sub

End Class";

            var expected = VerifyVB.Diagnostic("$saferootidentifiername$").WithLocation(0).WithArguments("TypeName");
            await VerifyVB.VerifyCodeFixAsync(test, expected, fixtest);
        }
    }
}

File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/VisualBasic/Diagnostic/Test/UnitTests.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports VerifyCS = Microsoft.CodeAnalysis.CSharp.Testing.MSTest.CodeFixVerifier(
    Of $saferootprojectname$.$saferootidentifiername$Analyzer,
    $saferootprojectname$.CSharp.CSharp$saferootidentifiername$CodeFixProvider)
Imports VerifyVB = Microsoft.CodeAnalysis.VisualBasic.Testing.MSTest.CodeFixVerifier(
    Of $saferootprojectname$.$saferootidentifiername$Analyzer,
    $saferootprojectname$.VisualBasic.VisualBasic$saferootidentifiername$CodeFixProvider)

Namespace $safeprojectname$
    <TestClass>
    Public Class $saferootidentifiername$UnitTest

        'No diagnostics expected to show up
        <TestMethod>
        Public Async Function TestMethod1_CSharp() As Task
            Dim test = ""
            Await VerifyCS.VerifyAnalyzerAsync(test)
        End Function

        <TestMethod>
        Public Async Function TestMethod1_VisualBasic() As Task
            Dim test = ""
            Await VerifyVB.VerifyAnalyzerAsync(test)
        End Function

        'Diagnostic And CodeFix both triggered And checked for
        <TestMethod>
        Public Async Function TestMethod2_CSharp() As Task

            Dim test = "
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    using System.Diagnostics;

    namespace ConsoleApplication1
    {
        class {|#0:TypeName|}
        {   
        }
    }"

            Dim fixtest = "
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    using System.Diagnostics;

    namespace ConsoleApplication1
    {
        class TYPENAME
        {   
        }
    }"

            Dim expected = VerifyCS.Diagnostic("$saferootidentifiername$").WithLocation(0).WithArguments("TypeName")
            Await VerifyCS.VerifyCodeFixAsync(test, expected, fixtest)
        End Function

        <TestMethod>
        Public Async Function TestMethod2_VisualBasic() As Task

            Dim test = "
Class {|#0:TypeName|}

    Sub Main()

    End Sub

End Class"

            Dim fixtest = "
Class TYPENAME

    Sub Main()

    End Sub

End Class"

            Dim expected = VerifyVB.Diagnostic("$saferootidentifiername$").WithLocation(0).WithArguments("TypeName")
            Await VerifyVB.VerifyCodeFixAsync(test, expected, fixtest)
        End Function
    End Class
End Namespace
